ynh_install_nodejs/ynh_install_nodejs

```diff
@@ -0,0 +1,301 @@
+# And propagate $PATH to sudo with $ynh_node_load_path
+# Exemple: `ynh_exec_as $app $ynh_node_load_path $ynh_npm install`
+#
+# $PATH contains the path of the requested version of Node.js.
+# However, $PATH is duplicated into $nodejs_path to outlast any manipulation of $PATH
+# You can use the variable `$ynh_node_load_path` to quickly load your Node.js version
+#  in $PATH for an usage into a separate script.
+# Exemple: $ynh_node_load_path $final_path/script_that_use_npm.sh`
+#
+#
+# Finally, to start a Node.js service with the correct version, 2 solutions
+#  Either the app is dependent of Node.js or npm, but does not called it directly.
+#  In such situation, you need to load PATH
+#    `Environment="__YNH_NODE_LOAD_PATH__"`
+#    `ExecStart=__FINALPATH__/my_app`
+#     You will replace __YNH_NODE_LOAD_PATH__ with $ynh_node_load_path
+#
+#  Or Node.js start the app directly, then you don't need to load the PATH variable
+#    `ExecStart=__YNH_NODE__ my_app run`
+#     You will replace __YNH_NODE__ with $ynh_node
+#
+#
+# one other variable is also available
+#   - $nodejs_path: The absolute path to Node.js binaries for the chosen version.
+#
+# usage: ynh_use_nodejs
+#
+# Requires YunoHost version 2.7.12 or higher.
+ynh_use_nodejs () {
+    nodejs_version=$(ynh_app_setting_get --app=$app --key=nodejs_version)
+
+    # Get the absolute path of this version of Node.js
+    nodejs_path="$nodejs_version_path/$YNH_APP_INSTANCE_NAME/bin"
+
+    # Allow alias to be used into bash script
+    shopt -s expand_aliases
+
+    # Create an alias for the specific version of Node.js and a variable as fallback
+    ynh_node="$nodejs_path/node"
+    alias ynh_node="$ynh_node"
+    # And npm
+    ynh_npm="$nodejs_path/npm"
+    alias ynh_npm="$ynh_npm"
+
+    # Load the path of this version of Node.js in $PATH
+    if [[ :$PATH: != *":$nodejs_path"* ]]; then
+        PATH="$nodejs_path:$PATH"
+    fi
+    # Create an alias to easily load the PATH
+    ynh_node_load_PATH="PATH=$PATH"
+    ynh_nodejs_load_path="PATH=$PATH"
+
+    # Sets the local application-specific Node.js version
+    pushd $final_path
+        $nodenv_install_dir/bin/nodenv local $nodejs_version
+    popd
+}
+
+# Install a specific version of Node.js
+#
+# ynh_install_nodejs will install the version of Node.js provided as argument by using nodenv.
+#
+# This helper creates a /etc/profile.d/nodenv.sh that configures PATH environment for nodenv
+# for every LOGIN user, hence your user must have a defined shell (as opposed to /usr/sbin/nologin)
+#
+# Don't forget to execute Node.js-dependent command in a login environment
+# (e.g. sudo --login option)
+# When not possible (e.g. in systemd service definition), please use direct path
+# to nodenv shims (e.g. $NODENV_ROOT/shims/bundle)
+#
+# usage: ynh_install_nodejs --nodejs_version=nodejs_version
+# | arg: -v, --nodejs_version= - Version of Node.js to install.
+#
+# Requires YunoHost version 2.7.12 or higher.
```

Comment:
Getops, used in this helper, requiers YunoHost 3.2.2
https://github.com/YunoHost/yunohost/blob/adc83b4c9c2c30e9ef75f3609c538b646f91f1db/data/helpers.d/getopts#L47

Reply:
updated :)